outputs/elasticsearch.go

```diff
@@ -105,6 +105,10 @@ func (c *Client) elasticsearchPost(index string, payload []byte, falcoPayloads .
 	c.EndpointURL = endpointURL
 
 	reqOpt := func(req *http.Request) {
+		if c.Config.Elasticsearch.ApiKey != "" {
+			req.Header.Set("Authorization", "APIKey "+c.Config.Elasticsearch.ApiKey)
+		}
+
 		if c.Config.Elasticsearch.Username != "" && c.Config.Elasticsearch.Password != "" {
 			req.SetBasicAuth(c.Config.Elasticsearch.Username, c.Config.Elasticsearch.Password)
 		}
```

Comment:
Should we not make them exclusive? APIKey or Username + Password but not both?

Reply:
I was think about this.

The original implementation of setting the headers 
https://github.com/falcosecurity/falcosidekick/blob/2.29.0/outputs/elasticsearch.go#L62
allowed the custom headers to overwrite the basic auth header. Effectively users could set the username and password in the configuration and then add "Authorization" header to the customHeaders and overwrite that header. It is a subtle detail, but it is the current behavior.
So I decided to follow the same behavior here, the headers are just applied in order, and not trying to figure out if the header is already set or not.

What is going to happen in this case: if the user has both apikey and the username and password in the configuration, the username and password will overwrite the "Authorization" header here:
https://github.com/golang/go/blob/ffb3e574012ce9d3d5193d7b8df135189b8a6671/src/net/http/request.go#L1023
thus will end up using basic auth. 

Let me know if you still want to change. 
And if we change, should we apply the custom headers before auth or after? There were applied after authroziation header before (and now still). If we continue to apply them after, should we check if the "Authorization" header was already set or let the user overwrite it if they want? If we let the user to overwrite, we might as well keep the implementation as already is.

Reply:
It's ok to me to keep the behavior like it is. Thanks